### PR TITLE
#170947435 Build sign-in feature

### PR DESCRIPTION
**Pull request by RusimbiPatrick**
### What does this PR do?

- Have sign-in feature build
- Have sign-in feature tested

### How should this be manually tested?

- Clone the [repo](https://github.com/andela/octopus-bn-backend.git)
- Run ``` npm install ```
-  Hit `http://localhost:3000/api/v1/auth/signin` to test it.

### Implementation plan

[Implementation plan name](https://drive.google.com/drive/u/1/folders/10O6cku3mfx2EQjTnghkYTr4tixe8xe-A)

### Stories associated with PR

[##170947435](https://www.pivotaltracker.com/story/show/170947435)

### Screenshots if any
Screen Shot 2020-02-12 at 13 10 11
]
**On a different type of user**
Screen Shot 2020-02-12 at 12 01 32

### CODE REVIEW

**Review by blaise82 — APPROVED** [review]

**Review by hakoemmy — APPROVED** [review]

**Review by izzett222 — APPROVED** [review]

File: src/controllers/user.controller.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/* eslint-disable require-jsdoc */
+import service from '../services/user.services';
```
Comment (kcharles52) [inline src/controllers/user.controller.js]:
```suggestion
import service from '../services/userServices';
```

File: src/database/migrations/20200206093213-create-users.js [hunk 2]
```diff
@@ -28,4 +40,4 @@ export default {
     }
   }),
   down: (queryInterface) => queryInterface.dropTable('Users')
-};
+};
```
Comment (kcharles52) [inline src/database/migrations/20200206093213-create-users.js]:
```suggestion
};

```

File: src/routes/api/user.routes.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import express from 'express';
+import userController from '../../controllers/user.controller';
```
Comment (kcharles52) [inline src/routes/api/user.routes.js]:
```suggestion
import userController from '../../controllers/userController';
```

File: src/routes/api/user.routes.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+
+const router = express.Router();
+
+
+router.post('/signin', userController.signin);
```
Comment (kcharles52) [inline src/routes/api/user.routes.js]:
```suggestion
router.post('/signin', userController.signin);

```

File: src/routes/api/user.routes.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import express from 'express';
+import userController from '../../controllers/user.controller';
+
+const router = express.Router();
+
+
```
Comment (kcharles52) [inline src/routes/api/user.routes.js:197]:
```suggestion
```

File: src/services/user.services.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
+/* eslint-disable require-jsdoc */
```
Comment (kcharles52) [inline src/services/user.services.js]:
```suggestion

```

File: src/services/user.services.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+import bcrypt from 'bcryptjs';
+import Models from '../database/models';
+// import serialize from '../utils/jwtTokenizer';
+
+
+class userService {
```
Comment (kcharles52) [inline src/services/user.services.js]: add documentation for the class and methods

File: src/services/user.services.js [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+      if (!registered) {
+        return {
+          status: 401,
+          error: 'Incorrect username or password combination',
+        };
+      }
+
+      const { method } = registered;
+      if (method !== 'local') {
```
Comment (kcharles52) [inline src/services/user.services.js]:
```suggestion
      
      if (method !== 'local') {
```

File: src/test/signin.test.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+        expect(res.body).to.have.keys('status', 'error');
+        expect(res.body.status).to.be.equal(401);
+        expect(res.body.error).to.be.equal('Incorrect username or password combination');
+        expect(res.body.error).to.be.a('string');
+        expect(res);
+        done();
+      });
+  });
+
+  it('It should show an error if the uses uses Open Auth', (done) => {
```
Comment (kcharles52) [inline src/test/signin.test.js]: check the grammar used in the test description

**Review by kcharles52 — CHANGES_REQUESTED** [review]

**Review by AbdoulNuru — APPROVED** [review]

**Review by kcharles52 — APPROVED** [review]

**Pull request merged**